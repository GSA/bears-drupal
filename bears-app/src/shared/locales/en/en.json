{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start",
      "list": [
        {
          "notice": "<div>Your information<strong>will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "heading": "Your potential benefits",
    "description": "<div>You must verify your eligibility and visit each agency to apply. <strong>Check with your city, state, or employer for other benefits.</strong></div>",
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    }
  },
  "notEligibleResults": {
    "heading": "Benefits you did not qualify for",
    "description": "<div>According to your answers you <strong>are not</strong> eligible for these benefits.</div>",
    "heading2": "Non-eligible Results",
    "description2": "Check with your state, city, or employer for other benefits."
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  }
}
