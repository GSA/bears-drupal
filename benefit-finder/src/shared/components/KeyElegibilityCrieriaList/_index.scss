#benefit-finder {
  .key-eligibility-criteria-group {
    .key-eligibility-criteria-heading {
      background-color: #1a4480;
      border-radius: 4px 4px 0 0;
      padding: 12px 36px;
      margin: 0;
      color: white;
      text-align: center;
    }

    .key-eligibility-criteria-list {
      padding: 0;
      margin: 0;

      .key-eligibility-criteria-list-item {
        margin: 0;
        padding: 14px 36px;
        border: 0.5px solid #005ea2;
        background-color: #d9e8f6;
      }
    }
  }
}
