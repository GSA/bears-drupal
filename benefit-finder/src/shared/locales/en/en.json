{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start:",
      "list": [
        {
          "notice": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "alertBanner": {
    "heading": "Error status",
    "description": "All fields marked required must be completed."
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "qualified": {
      "chevron": {
        "heading": "Your Custom List",
        "description": "<div><p><strong>Here is your list of potential Federal Benefits.</strong> Your state, city or employer may offer other benefits. Visit each agency providing the benefit to learn if you are truly eligible and conduct an actual application.</p></div>"
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you meet the criteria to qualify for these benefits:</strong></div>"
    },
    "notQualified": {
      "chevron": {
        "heading": "Benefits You Did Not Qualify For",
        "description": ""
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you did not meet the criteria to qualify for these benefits:</strong></div>"
    },
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    },
    "relativeBenefits": {
      "heading": "Other benefits that might be relevant to you"
    },
    "shareResults": {
      "heading": "Sharing and printing",
      "shareButton": "Share link",
      "emailButton": "Email results",
      "emailSubject": "Results from USAGovâ€™s benefit finder",
      "printButton": "Print results",
      "description": "Copy a link to this page with the criteria you selected above. Make sure to share it only with those you trust as your answers will be visible."
    }
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  },
  "sectionHeadings": {
    "start": "Complete Questions",
    "continue": "Questions Continued",
    "final": "Final Questions"
  },
  "select": {
    "labelSelect": "Select",
    "defaultValue": "Click to Select",
    "dateDefaultValue": "-Select-"
  },
  "date": {
    "alert": "Date should follow the format 01 - January 22 1977",
    "labelDay": "Day",
    "labelMonth": "Month",
    "labelYear": "Year",
    "monthOptions": [
      {
        "label": "01 - January"
      },
      {
        "label": "02 - February"
      },
      {
        "label": "03 - March"
      },
      {
        "label": "04 - April"
      },
      {
        "label": "05 - May"
      },
      {
        "label": "06 - June"
      },
      {
        "label": "07 - July"
      },
      {
        "label": "08 - August"
      },
      {
        "label": "09 - September"
      },
      {
        "label": "10 - October"
      },
      {
        "label": "11 - November"
      },
      {
        "label": "12 - December"
      }
    ]
  }
}
