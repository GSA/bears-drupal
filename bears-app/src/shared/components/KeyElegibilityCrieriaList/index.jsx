import { Heading } from '../index'
import PropTypes from 'prop-types'
import { useHandleClassName } from '../../hooks/useHandleClassName'
import './_index.scss'

const KeyElegibilityCrieriaList = ({
  className,
  data,
  initialEligibilityLength,
}) => {
  const defaultClasses = ['key-eligibility-criteria-group']
  return (
    <div className={useHandleClassName({ className, defaultClasses })}>
      {data && (
        <>
          {' '}
          <Heading
            className="key-eligibility-criteria-heading"
            headingLevel={5}
          >
            {`Key Eligibility Criteria Meet ${data.length} of
            ${initialEligibilityLength}`}
          </Heading>
          <ul className="key-eligibility-criteria-list">
            {data.map((item, index) => {
              const { criteriaKey, label } = item
              return (
                <li
                  key={`${index}-${criteriaKey}`}
                  className="usa-list usa-list--unstyled key-eligibility-criteria-list-item"
                >
                  {label}
                </li>
              )
            })}
          </ul>
        </>
      )}
    </div>
  )
}

KeyElegibilityCrieriaList.propTypes = {
  props: PropTypes.any,
}

export default KeyElegibilityCrieriaList
