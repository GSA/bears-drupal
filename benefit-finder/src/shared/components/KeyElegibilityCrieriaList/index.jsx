import { Heading } from '../index'
import PropTypes from 'prop-types'
import { useHandleClassName } from '../../hooks/useHandleClassName'
import './_index.scss'

/**
 * a functional component that renders a link as a button
 * @component
 * @param {string} className - inherited class names
 * @param {array} data - contains our criteria
 * @param {string} initialEligibilityLength - location
 * @return {html} returns a group with a heading and an unorderd list
 */
const KeyElegibilityCrieriaList = ({
  className,
  data,
  initialEligibilityLength,
  ui,
}) => {
  const { benefitSummary, benefitSummaryConjunction } = ui
  const defaultClasses = ['key-eligibility-criteria-group']

  return (
    <div className={useHandleClassName({ className, defaultClasses })}>
      {data && (
        <>
          {' '}
          <Heading
            className="key-eligibility-criteria-heading"
            headingLevel={5}
          >
            {`${benefitSummary} ${data.length} ${benefitSummaryConjunction}
            ${initialEligibilityLength}`}
          </Heading>
          <ul className="key-eligibility-criteria-list">
            {data.map((item, index) => {
              const { criteriaKey, label } = item
              return (
                <li
                  key={`${criteriaKey}-${index}`}
                  className="usa-list usa-list--unstyled key-eligibility-criteria-list-item"
                >
                  {label}
                </li>
              )
            })}
          </ul>
        </>
      )}
    </div>
  )
}

KeyElegibilityCrieriaList.propTypes = {
  className: PropTypes.string,
  data: PropTypes.array,
  initialEligibilityLength: PropTypes.number,
}

export default KeyElegibilityCrieriaList
