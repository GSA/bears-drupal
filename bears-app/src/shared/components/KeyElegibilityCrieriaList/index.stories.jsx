import KeyElegibilityCrieriaList from './index.jsx'

import * as DATA from '../../api/mock-data/content-data.json'

const data = DATA.benefits[0].benefit.eligibility
const initialEligibilityLength =
  DATA.benefits[0].benefit.initialEligibilityLength

export default {
  component: KeyElegibilityCrieriaList,
  tags: ['autodocs'],
  args: {
    data,
    initialEligibilityLength,
  },
}

export const Primary = {}
