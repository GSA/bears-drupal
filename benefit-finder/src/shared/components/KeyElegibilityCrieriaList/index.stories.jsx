import KeyElegibilityCrieriaList from './index.jsx'

import content from '../../api/mock-data/content-data.js'

const { lifeEventForm } = JSON.parse(content)
const data = lifeEventForm.benefits[0].benefit.eligibility
const initialEligibilityLength =
  lifeEventForm.benefits[0].benefit.initialEligibilityLength

export default {
  component: KeyElegibilityCrieriaList,
  tags: ['autodocs'],
  args: {
    data,
    initialEligibilityLength,
  },
}

export const Primary = {}
