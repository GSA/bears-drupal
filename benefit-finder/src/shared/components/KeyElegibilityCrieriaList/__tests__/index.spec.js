import renderer from 'react-test-renderer'
import KeyElegibilityCrieriaList from '../index.jsx'
import * as en from '../../../locales/en/en.json'

describe('KeyElegibilityCrieriaList', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(
      <KeyElegibilityCrieriaList ui={en.resultsView.benefitAccordion} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
