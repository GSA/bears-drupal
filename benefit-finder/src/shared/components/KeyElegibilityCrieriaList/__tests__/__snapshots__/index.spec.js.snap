// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyElegibilityCrieriaList renders a match to the previous snapshot 1`] = `
<div
  className=""
/>
`;
